sdk/core/core-http/src/policies/deserializationPolicy.ts

```diff
@@ -150,24 +150,55 @@ export function deserializeResponseBody(
     }
 
     const responseSpec = getOperationResponse(parsedResponse);
-    const expectedStatusCodes = Object.keys(operationSpec.responses);
-    const hasNoExpectedStatusCodes =
-      expectedStatusCodes.length === 0 ||
-      (expectedStatusCodes.length === 1 && expectedStatusCodes[0] === "default");
-    const isExpectedStatusCode: boolean = hasNoExpectedStatusCodes
-      ? 200 <= parsedResponse.status && parsedResponse.status < 300
-      : !!responseSpec;
-
-    // There is no operation response spec for current status code.
-    // So, treat it as an error case and use the default response spec to deserialize the response.
-    if (!isExpectedStatusCode) {
-      const defaultResponseSpec = operationSpec.responses.default;
-      if (!defaultResponseSpec) {
+
+    const statusCodes: string[] = Object.keys(operationSpec.responses);
```

Comment:
It would be awesome if you could extract everything from L154-L252 into another `handleErrorResponse` function. I think it will help a lot the readability of the deserialize function.

Reply:
@joheredi Done